pandas/tests/series/test_constructors.py

```diff
@@ -681,10 +681,12 @@ def test_constructor_dtype_datetime64(self):
         assert s.dtype == 'M8[ns]'
 
         # GH3414 related
-        msg = (r"cannot astype a datetimelike from \[datetime64\[ns\]\] to"
-               r" \[int32\]")
+        # msg = (r"cannot astype a datetimelike from \[datetime64\[ns\]\] to"
+        #        r" \[int32\]")
         # with pytest.raises(TypeError, match=msg):
-        Series(Series(dates).astype('int') / 1000000, dtype='M8[ms]')
+        #     Series(Series(dates).astype('int') / 1000000, dtype='M8[ms]')
+        pytest.raises(TypeError, lambda x: Series(
```

Comment:
this is raising `missing 1 required positional argument: 'x'` because of the lambda and not failing as intended. correcting this with the conversion to context manger is only failing on windows and again from the error message is probably not failing as intended

Reply:
this *should* be raising TypeError, what happens when you use the context manager

Reply:
it _is_ raising TypeError in master because of the lambda. this is a bad test.

correcting it with the context manager raises `TypeError: cannot astype a datetimelike from [datetime64[ns]] to [int32]` but only on the windows tests (see ci failures)

i've reverted the changes to the test, but the ci is failing on an unrelated conda error at the moment.

with the changes in this PR there will still be 823 uses of pytest.raises without the cm. so i don't want to get bogged down with bad tests at the moment.

Reply:
k that's fine (though happy to take your change, though it should raise on windows as well). can you create an issue about this.